cluster/gke_node_pools.go

```diff
@@ -63,13 +71,19 @@ func createNodePoolsFromClusterModel(clusterModel *google.GKEClusterModel) ([]*g
 	for i := 0; i < nodePoolsCount; i++ {
 		nodePoolModel := clusterModel.NodePools[i]
 
+		labels := map[string]string{
+			pkgCommon.LabelKey:         nodePoolModel.Name,
+			pkgCommon.OnDemandLabelKey: strconv.FormatBool(!nodePoolModel.Preemptible),
+		}
+
+		for _, label := range nodePoolModel.Labels {
+			labels[label.Name] = label.Value
```

Comment:
Return with error if user provided labels conflicts with Pipeline reserved labels

Reply:
Isn't it enough to check earlier in the `CreateClusterGKE.Validate()` function, [here](https://github.com/Laci21/pipeline/blob/create-gke-cluster-with-custom-labels/pkg/cluster/gke/gke.go#L119)?
(I'll modify that implementation to check for some kind of Banzai Cloud or Pipeline specific label name, instead of checking all labels one by one, once that pattern is decided.)